core/templates/domain/exploration/InteractionObjectFactorySpec.ts

```diff
@@ -18,40 +18,33 @@
 
 import { TestBed } from '@angular/core/testing';
 
-import { AnswerGroupObjectFactory } from
+import { AnswerGroupObjectFactory, IAnswerGroupBackendDict } from
   'domain/exploration/AnswerGroupObjectFactory';
 import { CamelCaseToHyphensPipe } from
   'filters/string-utility-filters/camel-case-to-hyphens.pipe';
-import { HintObjectFactory } from 'domain/exploration/HintObjectFactory';
-import { InteractionObjectFactory } from
+import { ConvertToPlainTextPipe } from
+  'filters/string-utility-filters/convert-to-plain-text.pipe';
+import { HintObjectFactory, IHintBackendDict } from
+  'domain/exploration/HintObjectFactory';
+import { IInteractionBackendDict, InteractionObjectFactory } from
   'domain/exploration/InteractionObjectFactory';
-import { OutcomeObjectFactory } from
+import { IOutcomeBackendDict, OutcomeObjectFactory } from
   'domain/exploration/OutcomeObjectFactory';
-import { SolutionObjectFactory } from
+import { ISolutionBackendDict, SolutionObjectFactory } from
   'domain/exploration/SolutionObjectFactory';
 
 describe('Interaction object factory', () => {
-  let iof = null;
-  let oof = null;
-  let agof = null;
-  let hof = null;
-  let sof = null;
-  let answerGroupsDict = null;
-  let defaultOutcomeDict = null;
-  let solutionDict = null;
-  let hintsDict = null;
-  let interactionDict = null;
-
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [CamelCaseToHyphensPipe]
+      providers: [CamelCaseToHyphensPipe, ConvertToPlainTextPipe]
     });
-    iof = TestBed.get(InteractionObjectFactory);
-    oof = TestBed.get(OutcomeObjectFactory);
-    agof = TestBed.get(AnswerGroupObjectFactory);
-    hof = TestBed.get(HintObjectFactory);
-    sof = TestBed.get(SolutionObjectFactory);
-    defaultOutcomeDict = {
+    this.iof = TestBed.get(InteractionObjectFactory);
```

Comment:
Why have you changed `let iof` to `this.iof`? 

(Just curious)

Reply:
We've been discussing this in an email thread, I'll add you in!

Reply:
Okay, thanks!